tests/integration/test_proxy.py

```diff
@@ -59,3 +89,23 @@ def test_use_proxy(tmpdir, httpbin, proxy_server):
         assert cassette_response.headers[key] == response.headers[key]
     assert cassette_response.headers == response.headers
     assert cassette.play_count == 1
+
+
+def test_use_https_proxy(tmpdir, httpbin_secure, proxy_server):
+    """Ensure that it works with an HTTPS proxy."""
+    with vcr.use_cassette(str(tmpdir.join("proxy.yaml"))):
+        response = requests.get(httpbin_secure.url, proxies={"https": proxy_server})
+
+    with vcr.use_cassette(str(tmpdir.join("proxy.yaml")), mode="once") as cassette:
+        cassette_response = requests.get(
+            httpbin_secure.url,
+            proxies={"https": proxy_server},
+        )
+
+    for key in set(cassette_response.headers.keys()) & set(response.headers.keys()):
+        assert cassette_response.headers[key] == response.headers[key]
```

Comment:
```suggestion
```
?

Seems to be fully covered by the upcoming line `assert cassette_response.headers == response.headers`. Am I missing something?

Reply:
You're probably right, I copied these assertions from `test_use_proxy` above. I assumed whoever wrote that might know more than I do. Should I clean it up there as well?

Reply:
@alga I think there are two ways to look at it:
- a) it is considered out of scope and is not fixed in here
- b) it is considered loosely related (or boy scout rule is applied) and then it nice to improve this finding elsewhere also *but* in separate commits so that it's clear that is not strongly related to the core matter of the pull request.

I'm good either way, please choose as you prefer.